android/java/org/chromium/chrome/browser/vpn/activities/VpnServerSelectionActivity.java

```diff
@@ -106,67 +134,47 @@ public void getItemOffsets(
-            finish();
-        }
-        return super.onOptionsItemSelected(item);
+    private void updateAutomaticSelection(boolean isAutomatic) {
+        mServerRegionList.setVisibility(isAutomatic ? View.GONE : View.VISIBLE);
+        mServersSectionText.setVisibility(isAutomatic ? View.GONE : View.VISIBLE);
     }
 
     @Override
     public void finishNativeInitialization() {
         super.finishNativeInitialization();
-        BraveVpnNativeWorker.getInstance().getAllServerRegions();
+        initVpnService();
         showProgress();
-    }
-
-    @Override
-    public void onGetAllServerRegions(String jsonResponse, boolean isSuccess) {
-        if (isSuccess) {
-            List<BraveVpnServerRegion> braveVpnServerRegions =
-                    BraveVpnUtils.getServerLocations(jsonResponse);
-            Collections.sort(
-                    braveVpnServerRegions,
-                    new Comparator<BraveVpnServerRegion>() {
-                        @Override
-                        public int compare(
-                                BraveVpnServerRegion braveVpnServerRegion1,
-                                BraveVpnServerRegion braveVpnServerRegion2) {
-                            return braveVpnServerRegion1
-                                    .getNamePretty()
-                                    .compareToIgnoreCase(braveVpnServerRegion2.getNamePretty());
-                        }
-                    });
-            mBraveVpnServerSelectionAdapter = new BraveVpnServerSelectionAdapter();
-            mBraveVpnServerSelectionAdapter.setVpnServerRegions(braveVpnServerRegions);
-            mBraveVpnServerSelectionAdapter.setOnServerRegionSelection(
-                    new OnServerRegionSelection() {
-                        @Override
-                        public void onServerRegionClick(BraveVpnServerRegion braveVpnServerRegion) {
-                            if (BraveVpnPrefUtils.getServerRegion()
-                                    .equals(braveVpnServerRegion.getName())) {
-                                Toast.makeText(
-                                                VpnServerSelectionActivity.this,
-                                                R.string.already_selected_the_server,
-                                                Toast.LENGTH_SHORT)
-                                        .show();
-                            } else {
-                                changeServerRegion(braveVpnServerRegion);
-                            }
-                        }
-                    });
-            mServerRegionList.setAdapter(mBraveVpnServerSelectionAdapter);
-            hideProgress();
-        } else {
-            Toast.makeText(
-                            VpnServerSelectionActivity.this,
-                            R.string.fail_to_get_server_locations,
-                            Toast.LENGTH_LONG)
-                    .show();
-        }
+        mServiceHandler.getAllRegions(
```

Comment:
`mServiceHandler` in theory can be null, we should probably double check it here